starlette_admin/contrib/beanie/helpers.py

```diff
@@ -0,0 +1,205 @@
+import functools
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+    get_args,
+    get_origin,
+)
+
+from beanie import Document, Link
+from beanie.odm.enums import SortDirection
+from mongoengine.queryset import Q as BaseQ  # noqa: N811
```

Comment:
Sorry I missed this earlier. How difficult would it be to decouple the query parser from mongoengine? I'm not very familiar with Beanie, but I was able to build a similar parser in Odmantic without relying on any external dependencies.